platform/server/http/HttpService.ts

```diff
@@ -23,38 +23,29 @@ export class HttpService implements CoreService {
   }
 
   async start() {
-    this.configSubscription = this.config$
-      .filter(() => {
-        if (this.httpServer.isListening()) {
-          // If the server is already running we can't make any config changes
-          // to it, so we warn and don't allow the config to pass through.
-          this.log.error(
-            'Received new HTTP config after server was started. ' +
-              'Config will **not** be applied.'
-          );
-          return false;
-        }
+    this.configSubscription = this.config$.subscribe(() => {
+      if (this.httpServer.isListening()) {
+        // If the server is already running we can't make any config changes
+        // to it, so we warn and don't allow the config to pass through.
+        this.log.warn(
+          'Received new HTTP config after server was started. ' +
+            'Config will **not** be applied.'
+        );
+      }
+    });
 
-        return true;
-      })
-      .switchMap(
-        config =>
-          new Observable<void>(() => {
-            this.httpServer.start(config);
-
-            return () => {
-              // TODO: This is async! :/
-              this.httpServer.stop();
-            };
-          })
-      )
-      .subscribe();
+    await this.httpServer.start(await this.config$.first().toPromise());
```

Comment:
If understand correctly we use `BehaviorSubject` for config, so `first()` should work fine even if start/stop server several times since `first` will return the latest value anyway.

Reply:
nitpick: Maybe extract `const config = await this.config$.first().toPromise()`? Just makes it a tad more readable.

Reply:
Yeah, agree ✔️